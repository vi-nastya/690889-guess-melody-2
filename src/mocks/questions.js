export const questions = [
  {
    type: `genre`,
    genre: `rock`,
    answers: [
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/ccCommunity/Chad_Crouch/Arps/Chad_Crouch_-_Elipsis.mp3`,
        genre: `instrumental`,
      },
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/none_given/Scott_Holmes/Scott_Holmes_-_Singles/Scott_Holmes_-_Driven_To_Success.mp3`,
        genre: `rock`,
      },
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Mr_Tachyon/Helping_Hands_ID_1202.mp3`,
        genre: `jazz`,
      },
      {
        src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/ccCommunity/Lobo_Loco/Vagabond/Lobo_Loco_-_05_-_The_First_Moment_ID_1016.mp3`,
        genre: `folk`,
      },
    ],
  },
  {
    type: `artist`,
    song: {
      artist: `Jim Beam`,
      src: `https://files.freemusicarchive.org/storage-freemusicarchive-org/music/no_curator/Mr_Tachyon/Helping_Hands_ID_1202.mp3`,
    },
    answers: [
      {
        picture: `http://placehold.it/134x134`,
        artist: `John Snow`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jack Daniels`,
      },
      {
        picture: `http://placehold.it/134x134`,
        artist: `Jim Beam`,
      },
    ],
  },
];
