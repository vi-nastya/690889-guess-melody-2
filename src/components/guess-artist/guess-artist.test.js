import React from 'react';
import renderer from 'react-test-renderer';

import GuessArtist from './guess-artist';
import questions from '../../mocks/questions.js';

it(`Guess artist question is rendered correctly`, () => {
  const tree = renderer
    .create(
        <GuessArtist
          question={questions[1]}
          onAnswer={() => {}}
        />
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
