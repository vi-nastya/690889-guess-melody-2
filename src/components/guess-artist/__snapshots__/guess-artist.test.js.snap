// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Guess artist question is rendered correctly 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
      href="#"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <svg
      className="timer"
      viewBox="0 0 780 780"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        className="timer__line"
        cx="390"
        cy="390"
        r="370"
        style={
          Object {
            "filter": "url(#blur)",
            "transform": "rotate(-90deg) scaleY(-1)",
            "transformOrigin": "center",
          }
        }
      />
    </svg>
    <div
      className="timer__value"
      xmlns="http://www.w3.org/1999/xhtml"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          disabled={true}
          onClick={[Function]}
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
    >
      <div
        className="artist"
        onClick={[Function]}
      >
        <input
          className="artist__input visually-hidden"
          id="answer-0"
          name="answer"
          type="radio"
          value="answer-0"
        />
        <label
          className="artist__name"
          htmlFor="answer-0"
        >
          <img
            alt="John Snow"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          John Snow
        </label>
      </div>
      <div
        className="artist"
        onClick={[Function]}
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value="answer-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="Jack Daniels"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Jack Daniels
        </label>
      </div>
      <div
        className="artist"
        onClick={[Function]}
      >
        <input
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          type="radio"
          value="answer-2"
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="Jim Beam"
            className="artist__picture"
            src="http://placehold.it/134x134"
          />
          Jim Beam
        </label>
      </div>
    </form>
  </section>
</section>
`;
